velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -215,15 +273,23 @@ VeloxToSubstraitExprConvertor::toSubstraitLiteral(
   ::substrait::Expression_Literal* substraitField =
       google::protobuf::Arena::CreateMessage<::substrait::Expression_Literal>(
           &arena);
-
-  VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
-      convertVectorValue,
-      vectorValue->type()->kind(),
-      arena,
-      vectorValue,
-      litValue,
-      substraitField);
-
+  if (vectorValue->isScalar()) {
+    VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+        convertVectorValue,
+        vectorValue->type()->kind(),
+        arena,
+        vectorValue,
+        litValue,
+        substraitField);
+  } else if (
+      auto constantVector =
+          std::dynamic_pointer_cast<ConstantVector<ComplexType>>(vectorValue)) {
+    complexVectorToLiteral(arena, constantVector, substraitField);
```

Comment:
Here constantVector may represent an array, a map or a struct, but complexVectorToLiteral handles only arrays.

Reply:
Yes, I think we can implement struct or map type in other PR next time. Here we add a check. https://github.com/facebookincubator/velox/pull/2450/files#diff-65defa30896670ec464dc213f57cf6a8066b5022cf6b862ccc33c332b4a25120R242-R245